tools/block-generator/generator/generate_test.go

```diff
@@ -229,17 +229,48 @@ func TestWriteRoundZero(t *testing.T) {
 func TestWriteRound(t *testing.T) {
 	partitiontest.PartitionTest(t)
 	g := makePrivateGenerator(t, 0, bookkeeping.Genesis{})
-	var data []byte
-	writer := bytes.NewBuffer(data)
-	g.WriteBlock(writer, 0)
-	g.WriteBlock(writer, 1)
-	var block rpcs.EncodedBlockCert
+
+	prepWriter := func() ([]byte, *bytes.Buffer, rpcs.EncodedBlockCert) {
+		var data []byte
+		writer := bytes.NewBuffer(data)
+		return data, writer, rpcs.EncodedBlockCert{}
+	}
+
+	// Initial conditions of g from makePrivateGenerator:
+	require.Equal(t, uint64(0), g.round)
+
+	data, writer, block := prepWriter()
+	err := g.WriteBlock(writer, 0)
+	require.NoError(t, err)
+
+	require.Equal(t, uint64(1), g.round)
+	protocol.Decode(data, &block)
+	require.Equal(t, block, rpcs.EncodedBlockCert{})
+
+	// WriteBlocks at the offset are special: they only advance the round
+	// the first time when called.
+	data, writer, block = prepWriter()
+	err = g.WriteBlock(writer, 0)
+	require.NoError(t, err)
+
+	require.Equal(t, uint64(1), g.round)
+	protocol.Decode(data, &block)
+	require.Equal(t, block, rpcs.EncodedBlockCert{})
```

Comment:
these are repeating lines 242-248?

Reply:
Intentionally so, except the slight difference in line 256 v. 240.  `Write()` is no longer idempotent when `round == g.offsetRound` so I do need some kind of repetition.